content/knowledge_base/flare/consensus/avalanche/specification.md

```diff
@@ -0,0 +1,225 @@
+In the case of a payment system and Avalanche, the Snowball logic used to determine a value is the process of deciding whether a transaction is valid.
+
+The protocol maintains the set of all known transactions using a dynamic, append-only [Directed Acyclic Graph](../../../glossary.md#Directed-Acyclic-Graph-(DAG)) (DAG) structure.
+There are two main positives from this:
+
+* DAG streamlines the path on which there are no conflicting transactions — a single vote on a DAG vertex implicitly votes for all transactions on the path to the genesis vertex
+* using the same principle as Bitcoin Network (BTC), the DAG tree renders past decisions on the path of transactions as more and more difficult to undo without approval of the correct nodes
+
+The genesis block in the DAG structure is represented by the graph first link, called "genesis vertex".
+This is the point where the first two edges of the graph meet.
+
+When a new transaction is created, its parents are defined in the DAG structure as its ancestors and the trascation is their newly created descendant.
+The parent-child relationship in DAG is not mandatory for a child transaction may not have any relation to its parents transactions outputs.
+It can actually spend funds received in other transactions from its [ancestor set](../../../glossary.md#Ancestor-Set).
+This is the set of all transactions reachable via parent edges throughout history.
+Oppositely — the term "progeny" means all existing (and potentially existing) children transactions and their children.
+
+The main purpose of a consensus protocol is to avoid the inclusion of conflicting transactions into the ledger.
+Even though the characteristics of a conflicting transaction are defined on an application level (ex. transactions that spend same UTXO), the notion of conflict can be abstracted in order to define the conflicting set.
+In Avalanche, every transaction belongs to a conflict set.
+This set consists of multiple transactions that are invalid towards each other or it is a singleton set - then it contains a single transaction (in the case of a virtuous transaction).
+Since reaching a consensus means to avoid any conflicts between transactions, only one transaction from each conflict set can be included in the approved path of transactions.
+
+Avalanche instantiates a Snowball instance for each conflict set.
+Avalanche treats the concept of repeated queries and multiple counters from Snowball taking advantage of the DAG structure.
+Specifically, when a transaction `T` is queried, all of its ancestors reachable through the DAG edges from it are implicitly part of the query.
+Consequently, nodes would respond positively to a query for transaction `T` only if `T` and all of its ancestors are the preferred option in their respective conflict sets.
+If more than a given threshold of responders vote positively, the transactions gets a [chit](../../../glossary.md#Chit).
```

Comment:
Yes it is the current transaction that gets a chit.
I am not sure what do you mean though - I see it is "gets" for the single entity, or you meant something else is wrong?

Reply:
It's just that in the current text it says "the transactionS getS a chit". Then the first one should be singular:
```suggestion
If more than a given threshold of responders vote positively, the transaction gets a [chit](../../../glossary.md#Chit).
```

It's essentially a nitpick, but also kind of confusing since to me as a noob it sounded like it might be that multiple transactions get one.

Reply:
Didn't see it, here is fixed https://github.com/optakt/optakt.github.io/pull/13/commits/79725b8076afee96d306b3736ea7fb3bef44c231#diff-2d4f16e70781857aec6689a748ad155e45611b81f184aee076bb139890410b98R42
Thanks!